Sources/Config.php

```diff
@@ -1178,22 +1181,24 @@ public static function reloadModSettings(): void
 			$integration_settings = Utils::jsonDecode(SMF_INTEGRATION_SETTINGS, true);
 
 			foreach ($integration_settings as $hook => $function) {
-				IntegrationHook::add($hook, $function, false);
+				if (is_string($function)) {
+					IntegrationHook::add($hook, $function, false);
+				} else {
+					IntegrationHook::register(
+						name: $hook,
+						function: $function['function'],
+						file: $function['file'],
+						class: $function['class'],
+						is_object: !empty($function['is_object']),
+						is_enabled: true,
+						permanent: false,
+					);
+				}
 			}
 		}
 
 		// Any files to pre include?
-		if (!empty(self::$modSettings['integrate_pre_include'])) {
-			$pre_includes = explode(',', self::$modSettings['integrate_pre_include']);
-
-			foreach ($pre_includes as $include) {
-				$include = strtr(trim($include), ['$boarddir' => self::$boarddir, '$sourcedir' => self::$sourcedir]);
-
-				if (file_exists($include)) {
-					require_once $include;
-				}
-			}
-		}
+		IntegrationHook::call('integrate_pre_include');
```

Comment:
This logic must be preserved.

Reply:
For an example,, see

https://github.com/live627/smf-custom-forms/blob/bf6527ba66bd53a7a1c60e6734deeab20c603c2b/src/package-info.xml#L50

Reply:
> This logic must be preserved.

@live627 , I believe the workaround I just sent up should resolve this.  I detect when we are calling an include hook and that `file` is empty, but `function` is not, then push `function` into `file`, so the logic is handled the same.
Should fix it to work the same and consistently.  Future calls to our hooks registration for `*_include` hooks should be with `function` empty and `file` specified.